daemon/graphdriver/devmapper/deviceset.go

```diff
@@ -841,11 +841,59 @@ func (devices *DeviceSet) createRegisterDevice(hash string) (*devInfo, error) {
 	return info, nil
 }
 
-func (devices *DeviceSet) createRegisterSnapDevice(hash string, baseInfo *devInfo, size uint64) error {
+func (devices *DeviceSet) takeSnapshot(hash string, baseInfo *devInfo, size uint64) error {
+	var doSuspend bool
+	deferredRemove := true
+
 	if err := devices.poolHasFreeSpace(); err != nil {
 		return err
 	}
 
+	if !devices.deferredRemove {
+		deferredRemove = false
+	}
+
+	devinfo, err := devicemapper.GetInfoWithDeferred(baseInfo.Name())
```

Comment:
GetInfoWithDeferred() works only if deferred removal support is there in libdm. So I think we will have to do following.

if devices.deferredRemove {
     devicemapper.GetInfoWithDeferred()
} else {
    devicemapper.GetInfo();
}